platform/posix/ota_pal/source/ota_pal_posix.c

```diff
@@ -323,6 +332,33 @@ static OtaPalStatus_t otaPal_CheckFileSignature( OtaFileContext_t * const C )
     return OTA_PAL_COMBINE_ERR( mainErr, 0 );
 }
 
+static void getFilePathFromCWD( char * pCompleteFilePath,
+                                const char * pFileName )
+{
+    char * pCurrentDir = NULL;
+
+    /* Get current directory. */
+    pCurrentDir = getcwd( pCompleteFilePath, OTA_FILE_PATH_LENGTH_MAX - 1 );
+
+    if( pCurrentDir == NULL )
+    {
+        LogError( ( "Failed to get current working directory: %s", strerror( errno ) ) );
+    }
+    else
+    {
+        /* Add the filename . */
+        if( strlen( pCompleteFilePath ) + strlen( pFileName ) + 2U > OTA_FILE_PATH_LENGTH_MAX )
+        {
+            LogError( ( "Insufficient space to generate file path" ) );
+        }
+        else
+        {
+            pCompleteFilePath[ strlen( pCompleteFilePath ) ] = '/';
```

Comment:
`pCompleteFilePath` is no longer null-terminated after this line, so the `strncat()` on the next line may produce indeterminate results.  Also, you should __not__ add 1 to the length in the `strncat()` on line 357.  The length is the maximum number of characters you want copied, __not including__ the null terminator which is added after the last character copied.  [`man strncat` for more details](https://en.cppreference.com/w/c/string/byte/strncat).  Please change lines 356 & 357 as follows:

```
strncat( pCompleteFilePath, "/" );
strncat( pCompleteFilePath, pFileName, strlen( pFileName ) );
```

Reply:
https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/4504c031ecd34a8fcbc2cbf40250b1b7be519792/platform/posix/ota_pal/source/ota_pal_posix.c#L359